summoner-cli/src/Summoner/Template/GitHub.hs

```diff
@@ -107,6 +109,76 @@ gitHubFiles Settings{..} = concat
         then ""
         else unlines ("\n# User specific" : settingsGitignore)
 
+    ghActionsYml :: Text
+    ghActionsYml = [text|
+        name: Haskell CI
+
+        on:
+          # Trigger the workflow on push or pull request,
+          # but only for the master branch
+          push:
+            branches:
+              - master
+          pull_request:
+
+        jobs:
+          build:
+            name: ghc ${{ matrix.ghc }}
+            runs-on: ubuntu-16.04
+            strategy:
+              matrix:
+                ghc: ${ghActionsVersions}
+                cabal: ["3.0"]
+
+            steps:
+            - uses: actions/checkout@v2
+              if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
+
+            - uses: actions/setup-haskell@v1
+              name: Setup Haskell
+              with:
+                ghc-version: ${{ matrix.ghc }}
+                cabal-version: ${{ matrix.cabal }}
+
+            # We cache the elements of the Cabal store separately,
+            # as the entirety of ~/.cabal can grow very large
+            # for projects with many dependencies.
+
+            - uses: actions/cache@v1
+              name: Cache ~/.cabal/packages
+              with:
+                path: ~/.cabal/packages
+                key: ${{ runner.os }}-${{ matrix.ghc }}-cabal-packages
+            - uses: actions/cache@v1
+              name: Cache ~/.cabal/store
+              with:
+                path: ~/.cabal/store
+                key: ${{ runner.os }}-${{ matrix.ghc }}-cabal-store
+            - uses: actions/cache@v1
+              name: Cache dist-newstyle
+              with:
+                path: dist-newstyle
+                key: ${{ runner.os }}-${{ matrix.ghc }}-fused-effects-dist
+
+            - name: Install dependencies
+              run: |
+                cabal new-update
+                cabal new-configure --enable-tests --write-ghc-environment-files=always -j2
+                cabal new-build --only-dependencies
```

Comment:
I went with the `new-` prefix because the `${cabalTest}` interpolation inserts `cabal new-test`, and an inconsistency in the generated source file kinda threw me off. I agree that the `new-` prefix isn’t ideal (it should either be omitted or specified precisely with `v2-`, but didn’t want to step outside the boundaries of this bug re. which Cabal version we use and which prefixes are used use in Travis/Appveyor/etc.

Thoughts? I don’t feel particularly strongly either way, so I’m happy to remove the prefixes if you like. I could also create a new interpolation rather than reusing `cabalTest`.

Reply:
Oh, I see. I'm okay with that then. We can change altogether when moving to Cabal-3.0 in all our settings.